api/workOrder/workOrderRouter.js

```diff
@@ -0,0 +1,494 @@
+      });
+  }
+);
+
+/**
+ * @swagger
+ * /workOrder:
+ *  post:
+ *    summary: Add a workOrder
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    requestBody:
+ *      description: WorkOrder object to to be added
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/WorkOrder'
+ *    responses:
+ *      400:
+ *        $ref: '#/components/responses/BadRequest'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'WorkOrder not found'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: workOrder created
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.post('/', authRequired, async (req, res) => {
+  const workOrder = req.body;
+  if (workOrder) {
+    const id = workOrder.id || 0;
+    try {
+      await WorkOrders.findById(id).then(async (pf) => {
+        if (pf == undefined) {
+          //workOrder not found so lets insert it
+          await WorkOrders.create(workOrder).then((workOrder) =>
+            res
+              .status(200)
+              .json({ message: 'workOrder created', workOrder: workOrder[0] })
+          );
+        } else {
+          res.status(400).json({ message: 'workOrder already exists' });
+        }
+      });
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ message: e.message });
+    }
+  } else {
+    res.status(404).json({ message: 'WorkOrder missing' });
+  }
+});
+/**
+ * @swagger
+ * /workOrder:
+ *  put:
+ *    summary: Update a workOrder
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    requestBody:
+ *      description: WorkOrder object to to be updated
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/WorkOrder'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: workOrder created
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.put('/', authRequired, function (req, res) {
+  const workOrder = req.body;
+  if (workOrder) {
+    const id = workOrder.id || 0;
+    WorkOrders.findById(id)
```

Comment:
While I believe you are doing a findById request first to check if it exists, it does not seem to actually be working.
I believe the cause for this is because you aren't actually checking what you get back from findById on line 422, adding a conditional should make it so you can actually see if it's a valid id or not

Reply:
I will check this and get back to you.

Reply:
Fixed with an additional check.